src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -929,6 +942,106 @@ private void DecodeModuleInitializerAttribute(DecodeWellKnownAttributeArguments<
             }
         }
 
+        private void DecodeInterceptsLocationAttribute(DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments)
+        {
+            Debug.Assert(arguments.AttributeSyntaxOpt is object);
+            Debug.Assert(!arguments.Attribute.HasErrors);
+            var attributeArguments = arguments.Attribute.CommonConstructorArguments;
+            if (attributeArguments is not [
+                { Kind: not TypedConstantKind.Array, Value: string filePath },
+                { Kind: not TypedConstantKind.Array, Value: int displayLineNumber },
+                { Kind: not TypedConstantKind.Array, Value: int displayCharacterNumber }])
+            {
+                // Since the attribute does not have errors (asserted above), it should be guaranteed that we have the above arguments.
+                throw ExceptionUtilities.Unreachable();
+            }
+
+            // Internally, line and character numbers are 0-indexed, but when they appear in code or diagnostic messages, they are 1-indexed.
+            int lineNumber = displayLineNumber - 1;
```

Comment:
>lineNumber [](http://example.com/codeflow?start=16&length=10)

Consider using a naming convention to distinguish between 0- and 1-indexed. For example: "Number"/"Position" for 1-indexed and "Index" for 0-indexed

Reply:
I wanted `display` prefix to serve this purpose. Would you prefer if both the "display" and "internal" numbers had a "name part" indicating which one they are?

Reply:
I see. Having a prefix or suffix on the "internal" numbers would be an improvement, but I wouldn't use "internal" (it would be hard to tell meaning). 
Consider "index" instead of "number" to indicate 0-indexed. Anyways, it's a nit.

Reply:
consider adding 'oneBased' to the name of anything internal that is in 1-based coordinates.

Reply:
Have adopted the OneBased/ZeroBased suffixes here.